### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by danielamoreno699**
Hello Review 

For this milestone project, I 

Create a method called current_user in your ApplicationController to make current user data available to all controllers.
It  returns the first user from the database as requested. Thus the post will be created for user one 

**Creates a Post on behalf of the current_user**
1. created a specific route for the new and create action 
2. created a PostForm controller to handle submission
3. created a view for postForm which displays a form ( the button "Add post" that displays in the USER INDEX will lead to route of the form)

image
image

**NOTE: When creating a post, it will always create a post for the first user of the database. in this case user 1 -(LILY)**

**Creates a Comment on behalf of the current_user**
1. created a specific route for the new and create action 
2. created a CommentForm controller to handle submission
3. created a view for CommentForm which displays a form ( the button "Add comment" that displays in the POST SHOW will lead to route of the form)

image
image

**NOTE: when creating a comment, as requested, the user who creates the comment is current_user( USER 1). however, the comment is registered in the  user session**

Add increment likes
1. Add a route for increment post likes

### CODE REVIEW

File: app/controllers/form_comment_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      render :new
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:Text)
+  end
+end
```
Comment (CollinsTatang) [inline app/controllers/form_comment_controller.rb]: **OPTIONAL**

- [ ] Please rename your file to `comments_controller.rb` and class `CommentController` for easy readability
    - Remember to update the file name change in your codebase with the new file name.

File: app/controllers/form_post_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      render :new
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:Title, :Text)
+  end
+end
```
Comment (CollinsTatang) [inline app/controllers/form_post_controller.rb]: **OPTIONAL**

- [ ] Please rename your file to `posts_controller.rb` and class `PostController` for easy readability
    - Remember to update the file name changes in your codebase with the new file names.

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class LikesController < ApplicationController
+  def increment
+    @post = Post.find(params[:id])
+    @like = Like.create(post: @post, user: current_user)
+
+    respond_to do |format|
+      format.js { render :increment }
+    end
+  end
```
Comment (CollinsTatang) [inline app/controllers/likes_controller.rb]: **OPTIONAL**

- [ ] Please update your function name from `increment` to `create` and remember to update your routes

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -12,4 +12,10 @@ def show
       format.json { render json: @post }
     end
   end
+
+  def increment_likes
+    @post = Post.find(params[:id])
+    @post.increment!(:LikesCounter)
+    redirect_to user_post_path(@post.author_id, @post.id)
+  end
```
Comment (CollinsTatang) [inline app/controllers/posts_controller.rb]: - [ ] Please we should have the create like method only inside the `LikeController`.
- [ ] Remember we should have only one controller to handle Post actions.

File: config/routes.rb [hunk 5]
```diff
@@ -3,9 +3,22 @@
...
     get '/users', to: 'users#index', as: 'users'
     get '/users/:id', to: 'users#show', as: 'user'
+
+    get '/form_post/new', to: 'form_post#new', as: 'new_form_post'
+    post '/form_post', to: 'form_post#create', as: 'form_posts'
+
+    get '/form_comment/new', to: 'form_comment#new', as: 'new_form_comment'
+    post '/form_comment', to: 'form_comment#create', as: 'form_comments'
+
+    get '/posts/:id/increment_likes', to: 'posts#increment_likes', as: 'increment_likes'
```
Comment (CollinsTatang) [inline config/routes.rb]: - [ ] Please remember to update your routes endpoints to match the controller file updates.

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @danielamoreno699, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DuaneDave — APPROVED** [review]
Hi @danielamoreno699 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Routes are correctly set up
✔️ Correct workflow used

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**